packages/react-core/src/filters/FilterQueryBuilder.js

```diff
@@ -17,39 +17,68 @@ export const getApplicableFilters = (filters = {}, owner) => {
   return filtersCopy;
 };
 
+// Filters to SQL
+
+const filterFunctions = {
+  [FilterTypes.IN](column, filterValues) {
+    const formattedValues = filterValues.map((v) => (isFinite(v) ? v : `'${v}'`));
+    return `${column} in(${formattedValues})`;
+  },
+  [FilterTypes.BETWEEN](column, filterValues) {
+    const queryFilters = filterValues.map(
+      ([left, right]) =>
+        `${left ? `${column} >= ${left}` : ``} ${left && right ? ' and ' : ''} ${
+          right ? `${column} <= ${right}` : ``
+        }`
+    );
+
+    return joinFilters(queryFilters);
+  },
+  [FilterTypes.TIME](column, filterValues) {
+    const formattedValues = filterValues.map((values) => [
+      `cast('${new Date(values[0]).toISOString()}' as timestamp)`,
+      `cast('${new Date(values[1]).toISOString()}' as timestamp)`
+    ]);
+    const tsColumn = `cast(${column} as timestamp)`;
+    const queryFilters = formattedValues.map(
+      ([left, right]) =>
+        `${left ? `${tsColumn} >= ${left}` : ``} ${left && right ? ' and ' : ''} ${
+          right ? `${tsColumn} <= ${right}` : ``
+        }`
+    );
+
+    return joinFilters(queryFilters);
+  },
+  [FilterTypes.CLOSED_OPEN](column, filterValues) {
+    const queryFilters = filterValues.map(
+      ([left, right]) =>
+        `${left ? `${column} >= ${left}` : ``} ${left && right ? ' and ' : ''} ${
+          right ? `${column} < ${right}` : ``
```

Comment:
I would say we have an issue with 0 treated as a falsy value vs undefined

Reply:
Yes, i'm fixing it